### PR TITLE
Rename Person and AddressBook

### PR DESCRIPTION
**Pull request by zhengyup**
Refactored
Persons -> InternTracker 
AddressBook -> InternTracker 
jsonAdaptedPerson -> jsonAdaptedInternApplication

related tests

**to do -> refactor other related files under storage

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -2,11 +2,11 @@ MIT License
 
 Copyright (c) 2016 Software Engineering Education - FOSS Resources
 
-Permission is hereby granted, free of charge, to any person obtaining a copy
+Permission is hereby granted, free of charge, to any internApplication obtaining a copy
```
Comment (tim-pipi) [inline LICENSE:5]: Hmmm, I don't think these should be changed?

File: LICENSE [hunk 2]
```diff
@@ -2,11 +2,11 @@ MIT License
...
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-copies of the Software, and to permit persons to whom the Software is
+copies of the Software, and to permit internApplications to whom the Software is
```
Comment (tim-pipi) [inline LICENSE:9]: Perhaps you should keep it as person for this as well?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -103,7 +103,7 @@ How the `Logic` component works:
 
 1. When `Logic` is called upon to execute a command, it is passed to an `AddressBookParser` object which in turn creates a parser that matches the command (e.g., `DeleteCommandParser`) and uses it to parse the command.
 1. This results in a `Command` object (more precisely, an object of one of its subclasses e.g., `DeleteCommand`) which is executed by the `LogicManager`.
-1. The command can communicate with the `Model` when it is executed (e.g. to delete a person).
+1. The command can communicate with the `Model` when it is executed (e.g. to delete a internApplication).
```
Comment (tim-pipi) [inline docs/DeveloperGuide.md:106]: Just a small nit - perhaps we could change this to `delete an internApplication`?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -103,7 +103,7 @@ How the `Logic` component works:
 
 1. When `Logic` is called upon to execute a command, it is passed to an `AddressBookParser` object which in turn creates a parser that matches the command (e.g., `DeleteCommandParser`) and uses it to parse the command.
```
Comment (tim-pipi) [inline docs/DeveloperGuide.md:104]: Should the AddressBook in `AddressBookParser` here be changed to InternTracker as well? 

I see this in other parts of the code as well!

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -175,11 +175,11 @@ Step 1. The user launches the application for the first time. The `VersionedAddr
 
 <puml src="diagrams/UndoRedoState0.puml" alt="UndoRedoState0" />
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+Step 2. The user executes `delete 5` command to delete the 5th internApplication in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
```
Comment (tim-pipi) [inline docs/DeveloperGuide.md:178]: Perhaps we could change the  addressBook here as well?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -633,17 +633,17 @@ testers are expected to do more *exploratory* testing.
 
 1. _{ more test cases …​ }_
 
-### Deleting a person
+### Deleting a internApplication
```
Comment (tim-pipi) [inline docs/DeveloperGuide.md:636]: Deleting an internApplication may be better?

File: docs/diagrams/ArchitectureSequenceDiagram.puml [hunk 7]
```diff
@@ -20,7 +20,7 @@ activate model MODEL_COLOR
 model -[MODEL_COLOR]-> logic
 deactivate model
 
-logic -[LOGIC_COLOR]> storage : saveAddressBook(addressBook)
+logic -[LOGIC_COLOR]> storage : saveAddressBook(internTracker)
```
Comment (tim-pipi) [inline docs/diagrams/ArchitectureSequenceDiagram.puml:23]: should this be saveInternTracker instead?

File: docs/tutorials/TracingCode.md [hunk 8]
```diff
@@ -195,22 +195,22 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
...
+       Person internApplicationToEdit = lastShownList.get(index.getZeroBased());
+       Person editedInternApplication = createEditedPerson(internApplicationToEdit, editPersonDescriptor);
+       if (!internApplicationToEdit.isSamePerson(editedInternApplication) && model.hasPerson(editedInternApplication)) {
            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
        }
-       model.setPerson(personToEdit, editedPerson);
+       model.setPerson(internApplicationToEdit, editedInternApplication);
        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
-       return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS, editedPerson));
+       return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS, editedInternApplication));
```
Comment (tim-pipi) [inline docs/tutorials/TracingCode.md:205]: Perhaps some of these constants may need some changes as well? I am referring to `MESSAGE_EDIT_PERSON_SUCCESS`?

I saw this in some other parts of the code as well!

File: docs/tutorials/TracingCode.md [hunk 9]
```diff
@@ -195,22 +195,22 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
...
-   * it uses the `setPerson()` method (defined in the interface `Model` and implemented in `ModelManager` as per the usual pattern) to update the person data.
-   * it uses the `updateFilteredPersonList` method to ask the `Model` to populate the 'filtered list' with _all_ persons.<br>
-     FYI, The 'filtered list' is the list of persons resulting from the most recent operation that will be shown to the user immediately after. For the `edit` command, we populate it with all the persons so that the user can see the edited person along with all other persons. If this was a `find` command, we would be setting that list to contain the search results instead.<br>
-     To provide some context, given below is the class diagram of the `Model` component. See if you can figure out where the 'filtered list' of persons is being tracked.
+   * it uses the `setPerson()` method (defined in the interface `Model` and implemented in `ModelManager` as per the usual pattern) to update the internApplication data.
+   * it uses the `updateFilteredPersonList` method to ask the `Model` to populate the 'filtered list' with _all_ internApplications.<br>
```
Comment (tim-pipi) [inline docs/tutorials/TracingCode.md:211]: Perhaps updateFilteredApplicationList here would be better as well?

File: docs/tutorials/TracingCode.md [hunk 10]
```diff
@@ -240,7 +240,7 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
      * {@code JsonSerializableAddressBook}.
      */
     public JsonSerializableAddressBook(ReadOnlyAddressBook source) {
-        persons.addAll(
+        internApplications.addAll(
             source.getPersonList()
```
Comment (tim-pipi) [inline docs/tutorials/TracingCode.md:244]: Would `getApplicationList()` be better?

File: src/main/java/seedu/address/MainApp.java [hunk 11]
```diff
@@ -15,15 +15,15 @@
 import seedu.address.commons.util.StringUtil;
 import seedu.address.logic.Logic;
 import seedu.address.logic.LogicManager;
-import seedu.address.model.AddressBook;
+import seedu.address.model.InternTracker;
 import seedu.address.model.Model;
 import seedu.address.model.ModelManager;
 import seedu.address.model.ReadOnlyAddressBook;
```
Comment (tim-pipi) [inline src/main/java/seedu/address/MainApp.java:21]: Should this be `ReadOnlyAddressBook` instead?

**Review by tim-pipi — COMMENTED** [review]
LGTM! Just some nits to pick about some missed `person` and `addressBook` reference still in the code!

File: LICENSE (hunk 1 above)
Comment (zhengyup) [reply LICENSE:5]: hahahahaha

**Comment by tim-pipi** [comment]
Also, perhaps we could rename this Pull Request to "Rename Person and AddressBook"

**Comment by zhengyup** [comment]
left the tutorials and sequence diagrams unchanged as we will probably change that later.

**Review by tim-pipi — APPROVED** [review]
LGTM!

**Review by AprupKale — COMMENTED** [review]
LGTM! 👍

**Pull request merged**